imap/index.c

```diff
@@ -5363,161 +5319,96 @@ extractor_timeout(struct protstream *s __attribute__((unused)),
                   struct prot_waitevent *ev __attribute__((unused)),
                   void *rock)
 {
-    struct backend *be = (struct backend *) rock;
+    struct extractor_ctx *ext = rock;
 
-    syslog(LOG_DEBUG, "extractor_timeout(%p)", be);
+    syslog(LOG_DEBUG, "extractor_timeout(%p)", ext);
 
     /* too long since we last used the extractor - disconnect */
-    extractor_disconnect(be);
+    extractor_disconnect(ext);
 
     return NULL;
 }
 
 static struct protocol_t http =
 { "http", "HTTP", TYPE_SPEC, { .spec = { &login, &ping, &logout } } };
 
-static int extractor_connect(const char *exturl, struct extractor_ctx *ext)
+static int extractor_connect(struct extractor_ctx *ext)
 {
     struct backend *be;
     struct backend_ctx *ctx = NULL;
-    struct buf buf = BUF_INITIALIZER;
-    const char *hostname;
-    char path[256];
     time_t now = time(NULL);
-    int r = 0;
 
-    syslog(LOG_DEBUG, "extractor_connect(%s)", exturl);
-
-    if (ext->failed > 1) {
-        syslog(LOG_INFO, "extractor failed for url: %s", exturl);
-        return IMAP_INTERNAL;
-    }
```

Comment:
We don't track failures any more - we'll try three times for every single attachment!